### PR TITLE
Fix translation and UI in Tavern dialog

### LINKED ISSUE #4795: Game, a save game in French language loaded in the program set in English language and continued to play let appear the text in French in the Tavern's rumor
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

In fheroes2, a save game in French language loaded in the program set in English language and continued to play let appear the text in French in the Tavern's rumor.

In fheroes2:

Tavern french 01

### Save file

Here is the save file:
[Tavern french 01.zip](https://github.com/ihhub/fheroes2/files/7727794/Tavern.french.01.zip)

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by ihhub**
- introduce new dynamic image UI element used for animations within a dialog
- introduce text UI element which can be used together with other UI elements
- fix incorrect translation of Tavern's rumors ( close #4795 )
- fix animation speed in Tavern's dialog

### CODE REVIEW

File: src/fheroes2/world/world.cpp [hunk 1]
```diff
@@ -798,11 +799,78 @@ void World::MonthOfMonstersAction( const Monster & mons )
...
+    case 4:
+        return _( "The end of the world is near." );
+    case 5:
+        return _( "The bones of Lord Slayer are buried in the foundation of the arena." );
+    case 6:
+        return _( "A Black Dragon will take out a Titan any day of the week." );
+    case 7:
+        return _( "He told her: Yada yada yada...  and then she said: Blah, blah, blah..." );
+    case 8:
+        return _( "An unknown force is being ressurected..." );
```
Comment (catadropa) [inline src/fheroes2/world/world.cpp:865]: Misspelled: it's resurrected

File: src/fheroes2/gui/ui_dialog.cpp [hunk 2]
```diff
@@ -629,4 +672,63 @@ namespace fheroes2
 
         showMessage( header, description, buttons, { this } );
     }
+
+    DynamicImageDialogElement::DynamicImageDialogElement( const int icnId, const std::vector<uint32_t> & backgroundIndecies, const uint64_t delay )
```
Comment (oleg-derevenetz) [inline src/fheroes2/gui/ui_dialog.cpp:676]: I believe there is a typo here and in other similar places: `backgroundIndecies` -> `backgroundIndices` or `backgroundIndexes`?

File: src/fheroes2/gui/ui_dialog.cpp [hunk 3]
```diff
@@ -629,4 +672,63 @@ namespace fheroes2
...
+        : _icnId( icnId )
+        , _backgroundIndecies( backgroundIndecies )
+        , _delay( delay )
+        , _currentIndex( 0 )
+    {
+        assert( !_backgroundIndecies.empty() );
+
+        assert( _delay > 0 );
```
Comment (oleg-derevenetz) [inline src/fheroes2/gui/ui_dialog.cpp:684]: We can join these assertions:

```cpp
assert( !_backgroundIndecies.empty() && _delay > 0 );
```

**Review by oleg-derevenetz — CHANGES_REQUESTED** [review]
Hi @ihhub I just left a few minor notes here, please take a look when possible.

File: src/fheroes2/world/world.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/world/world.cpp:865]: Great catch!

**Comment by ihhub** [comment]
Hi @oleg-derevenetz and @catadropa , I addressed your given comments. Please take a look again.

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**